test/Libraries/Microsoft.Extensions.AotCompatibility.TestApp/Microsoft.Extensions.AotCompatibility.TestApp.csproj

```diff
@@ -0,0 +1,50 @@
+<Project>
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>$(LatestTargetFramework)</TargetFramework>
+    <!-- clear TargetFrameworks because it is set in ..\Directory.Build.props -->
+    <TargetFrameworks></TargetFrameworks>
+    <PublishAot>true</PublishAot>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Remove the ReferenceTrimmer analyzer. We explicitly want to reference every assembly so the AOT compiler sees it. -->
+    <PackageReference Remove="ReferenceTrimmer" />
+
+    <LibraryProjects Include="$(RepoRoot)\src\Libraries\**\*.csproj" />
+    <!-- Azure.AI.Inference produces many warnings -->
+    <LibraryProjects Remove="$(RepoRoot)\src\Libraries\Microsoft.Extensions.AI.AzureAIInference\Microsoft.Extensions.AI.AzureAIInference.csproj" />
+    <!-- OpenAI produces a few warnings-->
+    <LibraryProjects Remove="$(RepoRoot)\src\Libraries\Microsoft.Extensions.AI.OpenAI\Microsoft.Extensions.AI.OpenAI.csproj" />
+    <!-- https://github.com/dotnet/extensions/issues/5624 -->
+    <LibraryProjects Remove="$(RepoRoot)\src\Libraries\Microsoft.Extensions.Caching.Hybrid\Microsoft.Extensions.Caching.Hybrid.csproj" />
+    <!-- https://github.com/dotnet/extensions/issues/5623 -->
+    <LibraryProjects Remove="$(RepoRoot)\src\Libraries\Microsoft.Extensions.Compliance.Redaction\Microsoft.Extensions.Compliance.Redaction.csproj" />
+    <!-- https://github.com/dotnet/extensions/issues/5625 -->
+    <LibraryProjects Remove="$(RepoRoot)\src\Libraries\Microsoft.Extensions.Http.Resilience\Microsoft.Extensions.Http.Resilience.csproj" />
+    
+    <TrimmerRootAssembly Include="@(LibraryProjects->'%(Identity)')" />
+    <ProjectReference Include="@(LibraryProjects)" />
+  </ItemGroup>
+
+  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+
+  <Target Name="PublishProject">
+    <!-- Always clean before publishing so we get warnings/errors every time. -->
+    <Exec Command="$(DOTNET_HOST_PATH) clean -c Release"
+          WorkingDirectory="$(MSBuildProjectDirectory)" />
+
+    <Exec Command="$(DOTNET_HOST_PATH) publish -c Release"
+          WorkingDirectory="$(MSBuildProjectDirectory)" />
+  </Target>
+
+  <!-- Override Test and VSTest so testing publishes this app and fails if there are any warnings/errors -->
+  <!-- Needs to be defined after importing the Sdk.targets -->
+  <Target Name="Test" DependsOnTargets="PublishProject" />
```

Comment:
Did you validate that this test actually runs as part of CI? I would assume so, but not sure if IsTestProject is requried.

Reply:
Yes, it runs as part of CI. In the previous commit I left one of the `LibraryProjects Remove` lines commented out to ensure it fails. See https://github.com/dotnet/extensions/runs/32829886244